tests/metadata_generator_test.py

```diff
@@ -0,0 +1,258 @@
+
+
+def test_create_dict_subelement_no_extra_data():
+    root = Element('root')
+    mg.create_dict_subelement(root, 'dog', 'Harold')
+    xml = tostring(root, pretty_print=True)
+    assert xml.decode('utf-8') == ('<root>\n'
+                                   '  <dog>Harold</dog>\n'
+                                   '</root>\n')
+
+
+def test_create_dict_subelement_degree_order_handling():
+    # Order should be name, level, discipline, grantor.
+    root = Element('root')
+    mg.create_dict_subelement(root,
+                              'degree',
+                              content={'discipline': 'Chemistry',
+                                       'level': 'Masters',
+                                       'grantor': 'UNT',
+                                       'name': 'foo'})
+    xml = tostring(root, pretty_print=True)
+    assert xml.decode('utf-8') == ('<root>\n'
+                                   '  <degree>\n'
+                                   '    <name>foo</name>\n'
+                                   '    <level>Masters</level>\n'
+                                   '    <discipline>Chemistry</discipline>\n'
+                                   '    <grantor>UNT</grantor>\n'
+                                   '  </degree>\n'
+                                   '</root>\n')
+
+
+def test_create_dict_subelement_content_dict_not_degree():
+    root = Element('root')
+    content = {'foo': 'one', 'bar': 'two'}
+    mg.create_dict_subelement(root, 'stuff', content=content)
+    # Check the content keys are children of the stuff element.
+    assert root[0].tag == 'stuff'
+    assert len(list(root[0])) == 2
+    for key, value in content.items():
+        assert root[0].find(key).text == value
+
+
+def test_highwiredict2xmlstring():
+    issue = hs.CitationIssue(content='1')
+    title = hs.CitationTitle(content='Important paper')
+    elements = [issue, title]
+    xml = mg.highwiredict2xmlstring(elements, ordering=['citation_title',
+                                                        'citation_issue'])
+    print(xml.decode('utf-8'))
+    assert xml.decode('utf-8') == ('<?xml version="1.0" encoding="UTF-8"?>\n'
+                                   '<metadata>\n'
+                                   '  <meta content="Important paper"'
+                                   ' name="citation_title"/>\n'
+                                   '  <meta content="1" name="citation_issue"/>\n'
+                                   '</metadata>\n')
+
+
+def test_breakString_shorter_than_width_with_offset():
+    line = mg.breakString('Hello world', width=79, firstLineOffset=0)
+    assert line == 'Hello world'
+
+
+def test_breakString_longer_than_width_with_offset():
+    line = mg.breakString('Hello world', width=10, firstLineOffset=4)
+    # Check line has been split to prevent surpassing width + offset length.
+    assert line == 'Hello\n world'
+
+
+def test_breakString_longer_than_width_with_offset_no_space():
```

Comment:
Since you did a test like this, I wonder if it would also be worth writing a test that has contiguous spaces where a line break would normally go. The function _only_ splits the line if there is a single space surrounded by two non-space characters.

Reply:
That might be a situation where we have an expected fail, if you do add it.

Reply:
I do think it is worth adding an `xfail` test for the case of multiple spaces--probably in most? cases, it would be desired to split on multiple spaces (unless maybe there is an endline char involved?)